src/node.js

```diff
@@ -22,16 +23,30 @@ export let internalStateKeys = {
   afterSearch: null,
 }
 
-export let initNode = (node, path, extend, types) => {
+export let autoKey = x => F.compactJoin('-', [x.field, x.type])
+
+export let initNode = (
+  node,
+  parentPath,
+  extend,
+  types,
+  dedupe = _.identity
+) => {
   runTypeFunction(types, 'init', node, extend)
-  extend(node, {
-    ..._.omit(_.keys(node), defaults),
-    ..._.omit(_.keys(node), getTypeProp(types, 'defaults', node)),
-    path,
-  })
+  if (node.key) dedupe(node.key) // add node.key to the dedupe cache
+  let key =
+    node.key ||
+    dedupe(runTypeFunctionOrDefault(autoKey, types, 'autoKey', node, extend))
+  extend(
+    node,
+    _.defaults(
+      { ...defaults, ...getTypeProp(types, 'defaults', node) },
+      { ...node, key, path: [...parentPath, key] }
```

Comment:
This change which gets rid of the `omit` and instead spreads node will cause it to trigger observable updates for every single property on every single node which is not ideal